### PR TITLE
add UnityEngine.InputSystem to YarnSpinner.Unity.Example.asmdef

### PR DESCRIPTION
**Pull request by radiatoryang**
tl;dr when I updated some Samples to be compatible with InputSystem, I forgot to add UnityEngine.InputSystem to the YarnSpinner.Unity.Example.asmdef

* **Please check if the pull request fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated to describe this change

To update the documentation on [yarnspinner.dev](https://yarnspinner.dev), please visit the [documentation repository](https://github.com/YarnSpinnerTool/Docs).

* **What kind of change does this pull request introduce?**

- [x] Bug Fix
- [ ] Feature
- [ ] Something else

* **What is the current behavior?** (You can also link to an open issue here)

this ends up breaking automated CI compiles under specific (and rare) circumstances:
1. import YarnSpinner as a local package via Git submodule
2. CI imports all Samples even if it's inside \Samples~\
3. compile fails because Samples can't find UnityEngine.InputSystem

* **What is the new behavior (if this is a feature change)?**

add UnityEngine.InputSystem to the YarnSpinner.Unity.Example.asmdef

* **Does this pull request introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

the original issue didn't affect most users I think, since either (1) Unity ignores the \Samples~\ folder (which is why I didn't catch the bug before?) or (2) users import Samples subfolders individually, so wouldn't Unity ignore the YarnSpinner.Unity.Example.asmdef in the root of the Samples folder anyway?

if the user doesn't have InputSystem package imported, the .asmdef reference will be null and it'll be ignored

* **Other information**:

sorry

### CODE REVIEW

**Comment by desplesda** [comment]
What impact does this have on users running earlier versions of Unity, such as Unity 2018?

**Comment by radiatoryang** [comment]
to review what I think the issue is:
- in rare circumstances with certain YS installs, the Unity Editor imports the entire \Samples\ folder and processes the .asmdef there
- and if the user also has InputSystem installed (so ENABLE_INPUT_SYSTEM define has been set), the Sample .cs files with `#if ENABLE_INPUT_SYSTEM` try to import a UnityEngine.InputSystem namespace...
- ... but compile fails because the .asmdef file isn't linking to UnityEngine.InputSystem

So what's different in Unity 2018 with all that? Does it handle Samples differently? I think asmdefs were added in 2017...

**Comment by desplesda** [comment]
The main difference is that the InputSystem package is only available in 2019.4 and later. I'll do a test of this branch to see what happens on 2018 when all of the `Samples~` directory is imported. Will report back!

**Comment by desplesda** [comment]
Seems to work fine - Unity 2018.4 will note that it can't find the assembly, and otherwise ignore it. The `ENABLE_INPUT_SYSTEM` define isn't defined, so no code is compiled that references the missing assembly. Happy to merge this now!
 
image
